FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheckAvailability.h

```diff
@@ -46,29 +46,9 @@
 
 #pragma mark - App Attest
 
-// App Attest availability was extended to macOS and Mac Catalyst in Xcode 12.5.
-#if (defined(__IPHONE_14_5) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_14_5) || \
-    (defined(__MAC_11_3) && __MAC_OS_X_VERSION_MAX_ALLOWED >= __MAC_11_3) ||        \
-    (defined(__TVOS_14_5) && __TV_OS_VERSION_MAX_ALLOWED >= __TVOS_14_5)
-
-// Targets where `DCAppAttestService` is available to be used in preprocessor conditions.
-#define FIR_APP_ATTEST_SUPPORTED_TARGETS TARGET_OS_IOS || TARGET_OS_OSX
-
-// `AppAttestProvider` availability annotations
-#define FIR_APP_ATTEST_PROVIDER_AVAILABILITY \
-  API_AVAILABLE(macos(11.0), ios(14.0)) API_UNAVAILABLE(tvos, watchos)
-
-#else  // (defined(__IPHONE_14_5) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_14_5) || \
-          (defined(__MAC_11_3) && __MAC_OS_X_VERSION_MAX_ALLOWED >= __MAC_11_3) ||        \
-          (defined(__TVOS_14_5) && __TV_OS_VERSION_MAX_ALLOWED >= __TVOS_14_5)
-
 // Targets where `DCAppAttestService` is available to be used in preprocessor conditions.
-#define FIR_APP_ATTEST_SUPPORTED_TARGETS TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+#define FIR_APP_ATTEST_SUPPORTED_TARGETS TARGET_OS_IOS || TARGET_OS_OSX || TARGET_OS_TV
 
 // `AppAttestProvider` availability annotations
 #define FIR_APP_ATTEST_PROVIDER_AVAILABILITY \
-  API_AVAILABLE(ios(14.0)) API_UNAVAILABLE(macos, tvos, watchos)
-
-#endif  // (defined(__IPHONE_14_5) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_14_5) || \
-          (defined(__MAC_11_3) && __MAC_OS_X_VERSION_MAX_ALLOWED >= __MAC_11_3) ||        \
-          (defined(__TVOS_14_5) && __TV_OS_VERSION_MAX_ALLOWED >= __TVOS_14_5)
+  API_AVAILABLE(macos(11.0), ios(14.0), tvos(15.0)) API_UNAVAILABLE(watchos)
```

Comment:
Sorry if this is a dumb question, do we need an annotation for `macCatalyst` now that `&& !TARGET_OS_MACCATALYST` has been removed or is it determined implicitly based on the `macos` and `ios` values for `API_AVAILABLE(...)`?

Reply:
Hmm good question. I don't think we need to. 

If you look at the `TargetConditionals.h` header, it has a nice diagram:
```
 *    +---------------------------------------------------------------------------+
 *    |                             TARGET_OS_MAC                                 |
 *    | +-----+ +-------------------------------------------------+ +-----------+ |
 *    | |     | |                  TARGET_OS_IPHONE               | |           | |
 *    | |     | | +-----------------+ +----+ +-------+ +--------+ | |           | |
 *    | |     | | |       IOS       | |    | |       | |        | | |           | |
 *    | | OSX | | | +-------------+ | | TV | | WATCH | | BRIDGE | | | DRIVERKIT | |
 *    | |     | | | | MACCATALYST | | |    | |       | |        | | |           | |
 *    | |     | | | +-------------+ | |    | |       | |        | | |           | |
 *    | |     | | +-----------------+ +----+ +-------+ +--------+ | |           | |
 *    | +-----+ +-------------------------------------------------+ +-----------+ |
 *    +---------------------------------------------------------------------------+
```

MacCatalyst is a subset of iOS so I typically don't include it unless I'm excluding it.

As for the `API_AVAILABLE` annotation, it doesn't look like we have any precedent in the codebase for using it. So I guess it is implicitly derived based on the iOS version. In the case of App Attest support, iOS the macCatalyst support versions are in sync (iOS/macCatalyst 14.0+).

Reply:
That's super helpful, thank you!

Reply:
Marking as resolved to merge, but feel free to comment and I can address any feedback in next PR!